app/controllers/bookings_controller.rb

```diff
@@ -1,10 +1,23 @@
 class BookingsController < ApplicationController
   skip_before_action :authenticate_user!
 
-
   def new
+    @jewel = Jewel.find(params[:jewel_id])
+    @user = current_user
+    @booking = Booking.new
   end
 
   def create
+    @jewel = Jewel.find(params[:jewel_id])
+    @booking = Booking.new(booking_params)
+    @user = current_user
+    @booking.user = @user
+    @booking.jewel = @jewel
+    @booking.save!
+    redirect_to jewel_path(@jewel), notice: "Booking validated"
```

Comment:
Would have done:
if @booking.save! 
redirect_to jewel_path(@jewel), notice: "Booking validated"
else 
render :new, status: :unprocessable_entity

The rest seems good